add shared_configs:update to cap restart

drh-stanford opened the pull request:
This PR adds `shared_configs:update` as a precursor to `deploy:restart`. Also, fixes the JAR file downloads to happen before, not after, a restart.

The PR updates dlss-capistrano to 3.4 in order to get the shared_configs stuff.

This PR fixes #26 .

ndushay commented on Capfile:
```diff
@@ -20,6 +20,7 @@ require 'capistrano/deploy'
 require 'capistrano/bundler'
 # require 'capistrano/rails/assets'
 # require 'capistrano/rails/migrations'
+require 'capistrano/shared_configs'
```
isn't this part of dlss/capistrano now?

ndushay commented on Gemfile:
```diff
@@ -29,5 +29,6 @@ end
 group :deployment do
   gem 'capistrano'
   gem 'capistrano-bundler'
+  gem 'capistrano-shared_configs'
```
I believe this is in dlss-capistrano ...

drh-stanford commented:
@ndushay i updated dlss-capistrano and removed the unneeded shared_configs inclusions. ready for review again. (btw i dunno how to acknowledge your github review -- it just says "Dismiss review" which doesn't sound right)
